### PR TITLE
chore(esm-samples): bump to 4.29 and update deps

### PR DESCRIPTION
**Pull request by andygup**
Notable changes:
- Added an OAuth ESM sample (thanks @lboyd93!) 
- Removed deprecated custom widget sample. This was replaced by jsapi-custom-ui sample.
- Updated deprecation notices for AMD local build and AMD TypeScript samples . This does not affect AMD CDN.

### CODE REVIEW

**Review by dasa — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -20,17 +20,17 @@ Below are a collection of samples that provide _proof-of-concept_ for using `@ar
 * [Vite/TypeScript](./esm-samples/jsapi-vite-ts/)
 * [Webpack (no framework)](./esm-samples/webpack/)
 
-## AMD TypeScript declaration files
+## HTML OAuth callback example
```
[the remark this answers is not in the document]
Comment (andygup) [reply README.md:23]: Good point, how about this for the title: "HTML OAuth callback"?

[the remark this answers is not in the document]
Comment (dasa) [reply README.md:23]: Sounds good, and it's also referred to as an example in the text.

[the remark this answers is not in the document]
Comment (andygup) [reply README.md]: Thanks. Updated.

[the remark this answers is not in the document]
Comment (andygup) [reply README.md]: Thanks. Updated.

File: README.md [hunk 2]
```diff
@@ -14,23 +14,24 @@ Below are a collection of samples that provide _proof-of-concept_ for using `@ar
...
 * [Rollup.js (no framework)](./esm-samples/rollup/)
 * [Vue.js/Vite](./esm-samples/jsapi-vue/)
 * [Vite/TypeScript](./esm-samples/jsapi-vite-ts/)
 * [Webpack (no framework)](./esm-samples/webpack/)
 
-## AMD TypeScript declaration files
+## HTML OAuth callback
 
-The TypeScript `.d.ts` files for the last several releases of the Maps SDK's AMD modules can be downloaded from this [repo](./4.x/typescript/README.md). Or, you can NPM install the [`arcgis-js-api`](https://www.npmjs.com/package/arcgis-js-api) package and the TypeScript definitions are included.
+If you have set OAuthInfo's `popup` property to `true`, you'll need to host a [callback page](./oauth/) on your server. For more information, visit the Maps SDK's [Access secure resources(https://developers.arcgis.com/javascript/latest/secure-resources/)] guide topic.
```
Comment (lboyd93) [inline README.md:26]: Might be good to add a link to the property.

```suggestion
If you have set OAuthInfo's [`popup`](https://developers.arcgis.com/javascript/latest/api-reference/esri-identity-OAuthInfo.html#popup) property to `true`, you'll need to host a [callback page](./oauth/) on your server. For more information, visit the Maps SDK's [Access secure resources(https://developers.arcgis.com/javascript/latest/secure-resources/)] guide topic.
```

File: esm-samples/jsapi-oauth/README.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# Accessing ArcGIS Online using OAuth 2.0 with ESM + Vite
+
+This repo demonstrates how to log in to ArcGIS Online or Enterprise using OAuth 2.0 authentication and the [IdentityManager](https://developers.arcgis.com/javascript/latest/api-reference/esri-identity-IdentityManager.html) with [@arcgis/core](https://www.npmjs.com/package/@arcgis/core) ES modules with [Vite](https://vitejs.dev/) .
```
Comment (lboyd93) [inline esm-samples/jsapi-oauth/README.md:3]:
```suggestion
This sample demonstrates how to log in to ArcGIS Online or Enterprise using OAuth 2.0 authentication and the [IdentityManager](https://developers.arcgis.com/javascript/latest/api-reference/esri-identity-IdentityManager.html) with [@arcgis/core](https://www.npmjs.com/package/@arcgis/core) ES modules with [Vite](https://vitejs.dev/) .
```

**Review by lboyd93 — APPROVED** [review]
LGTM! Left a couple of suggestions.

**Review by odoe — APPROVED** [review]
Looks like a winner

**Pull request merged**